### PR TITLE
(SERVER-812) Added auth.conf compatibility documentation

### PR DESCRIPTION
**Pull request by camlow325**
This commit includes the following documentation changes:

1) Adds a section to the `configuration.markdown` file, comparing some
legacy Puppet auth.conf settings to corresponding
trapperkeeper-authorization auth.conf settings.

2) Adds some distinctions related to the use of
trapperkeeper-authorization to the
`compatibility_with_puppet_agent.markdown` doc.

### CODE REVIEW

**Comment by camlow325** [comment]
fyi, @gguillotte.

File: documentation/configuration.markdown [hunk 1]
```diff
@@ -261,6 +265,155 @@ If you want to add a rule but let the default rules from Puppet take
 precedence over your new rule, you should use a `sort-order` for the rule
 which is in the range from 601 to 998 (inclusive).
 
+#### Legacy Puppet auth.conf compatibility considerations
+
+In general, the functionality provided for `trapperkeeper-authorization`
+"auth.conf"  rules is comparable to rule definitions with the core
+[Puppet auth.conf][auth.conf] file.  This section details some differences
+around rule definitions to keep in mind when migrating custom rules to the
+`trapperkeeper-authoration` "auth.conf" format.
```
Comment (nwolfe) [inline documentation/configuration.markdown]: s/authoration/authorization

**Comment by nwolfe** [comment]
:+1: just the one typo

File: documentation/configuration.markdown (hunk 1 above)
Comment (camlow325) [inline documentation/configuration.markdown]: Thanks.  Just fixed that up and squashed into 2b6e2c7.

**Comment by gguillotte** [comment]
I don't have any objections. As previously discussed the plan is for me to follow up with a comprehensive PR with edits to these docs and integration with related docs in puppet-docs.

**Comment by camlow325** [comment]
Sounds great.  Thanks for reviewing this, @gguillotte.

@nwolfe - you good with merging this, then?

**Pull request merged**